packages/cli/src/aws.ts

```diff
@@ -158,3 +218,180 @@ function printStackDetails(details: MedplumStackDetails): void {
+/**
+ * Uploads the app to S3.
+ * Ensures correct content-type and cache-control for each file.
+ * @param tmpDir The temporary directory where the app is located.
+ * @param bucketName The destination S3 bucket name.
+ */
+async function uploadAppToS3(tmpDir: string, bucketName: string): Promise<void> {
+  // Manually iterate and upload files
+  // Automatic content-type detection is not reliable on Microsoft Windows
+  // So we explicitly set content-type
+  const uploadPatterns: [string, string, boolean][] = [
+    // Cached
+    // These files generally have a hash, so they can be cached forever
+    // It is important to upload them first to avoid broken references from index.html
+    ['css/**/*.css', 'text/css', true],
+    ['css/**/*.css.map', 'application/json', true],
+    ['img/**/*.png', 'image/png', true],
+    ['img/**/*.svg', 'image/svg+xml', true],
+    ['js/**/*.js', 'application/javascript', true],
+    ['js/**/*.js.map', 'application/json', true],
+    ['js/**/*.txt', 'text/plain', true],
+    ['favicon.ico', 'image/vnd.microsoft.icon', true],
+    ['robots.txt', 'text/plain', true],
+    ['workbox-*.js', 'application/javascript', true],
+    ['workbox-*.js.map', 'application/json', true],
+
+    // Not cached
+    ['manifest.webmanifest', 'application/manifest+json', false],
+    ['service-worker.js', 'application/javascript', false],
+    ['service-worker.js.map', 'application/json', false],
+    ['index.html', 'text/html', false],
+  ];
+  for (const uploadPattern of uploadPatterns) {
+    await uploadFolderToS3({
+      rootDir: tmpDir,
+      bucketName,
+      fileNamePattern: uploadPattern[0],
+      contentType: uploadPattern[1],
+      cached: uploadPattern[2],
+    });
+  }
+}
+
+/**
+ * Uploads a directory of files to S3.
+ * @param options The upload options such as bucket name, content type, and cache control.
+ */
+async function uploadFolderToS3(options: {
+  rootDir: string;
+  bucketName: string;
+  fileNamePattern: string;
+  contentType: string;
+  cached: boolean;
+}): Promise<void> {
+  const items = fastGlob.sync(options.fileNamePattern, { cwd: options.rootDir });
+  for (const item of items) {
+    await uploadFileToS3(join(options.rootDir, item), options);
+  }
+}
+
+/**
+ * Uploads a file to S3.
+ * @param filePath The file path.
+ * @param options The upload options such as bucket name, content type, and cache control.
+ */
+async function uploadFileToS3(
+  filePath: string,
+  options: {
+    rootDir: string;
+    bucketName: string;
+    contentType: string;
+    cached: boolean;
+  }
+): Promise<void> {
+  const fileStream = createReadStream(filePath);
+  const s3Key = filePath
+    .substring(options.rootDir.length + 1)
+    .split(sep)
+    .join('/');
+
+  const putObjectParams = {
+    Bucket: options.bucketName,
+    Key: s3Key,
+    Body: fileStream,
+    ContentType: options.contentType,
+    CacheControl: options.cached ? 'public, max-age=31536000' : 'no-cache, no-store, must-revalidate',
```

Comment:
@mattwiller I think this fixes the `Cache-Control` issue that we discussed the other day.